openrr-client/src/clients/ik_client.rs

```diff
@@ -302,7 +304,9 @@ pub fn create_ik_client(
         k::Chain::from_end(full_chain.find(&config.ik_target).unwrap())
     };
     let arm_ik_solver_with_chain = create_ik_solver_with_chain(&full_chain, &config);
-    IkClient::new(client, arm_ik_solver_with_chain, arm_chain)
+    let mut client = IkClient::new(client, arm_ik_solver_with_chain, arm_chain);
+    client.constraints = constraints;
```

Comment:
Comment: I think it's better to add new constructor IkClient::new_with_constraints(a, b, c);.